core/node/libp2p/routing.go

```diff
@@ -179,3 +184,80 @@ func PubsubRouter(mctx helpers.MetricsCtx, lc fx.Lifecycle, in p2pPSRoutingIn) (
 		},
 	}, psRouter, nil
 }
+
+func AutoRelayFeeder(cfgPeering config.Peering) func(fx.Lifecycle, host.Host, chan peer.AddrInfo, *ddht.DHT) {
+	return func(lc fx.Lifecycle, h host.Host, peerChan chan peer.AddrInfo, dht *ddht.DHT) {
+		ctx, cancel := context.WithCancel(context.Background())
+		done := make(chan struct{})
+
+		defer func() {
+			if r := recover(); r != nil {
+				fmt.Println("Recovering from unexpected error in AutoRelayFeeder:", r)
+				debug.PrintStack()
+			}
+		}()
```

Comment:
I got paranoid because of panics around go-libp2p 0.19 (e.g. https://github.com/libp2p/go-libp2p/pull/1467), and since this is bolt-on temporary feeder (that may get refactored multiple times) felt prudent to decrease blast radius.

Reply:
seems like overkill but okay :)